File: Convertor de Temperatura C++/src/celsius.cpp
#include "celsius.hpp"

void Celsius::converte(float celsius)
{
    cout << "Temperatura em Celsius: " << setiosflags (ios::fixed) << setprecision (1) << celsius << " ºC" << endl;
    cout << "Temperatura em Fahrenheit: " << setiosflags (ios::fixed) << setprecision (1) << celsius * 9 / 5 + 32 << " ºF" << endl;
    cout << "Temperatura em Kelvin: " << setiosflags (ios::fixed) << setprecision (1) << celsius + 273 << " K" << endl;
}

File: Convertor de Temperatura Java/src/dados/Celsius.java
package dados;

public class Celsius extends Termometro
{

	public Celsius(double temperatura)
	{
		super(temperatura);
	}
	
	public void converte()
	{
		double celsius = getTemperatura();
		setCelsius(celsius);
		double fahrenheit = getTemperatura() * 9 / 5 + 32;
		setFahrenheit(fahrenheit);
		double kelvin = getTemperatura() + 273;
		setKelvin(kelvin);
	}

}
